app/src/main/java/com/example/rholbrook/tickettoride/game/GameMapFragmentPresenter.java

```diff
@@ -53,7 +53,8 @@ public void routeClaimed(Player player, Route route) {
     public void updateAvailableRoutes(List<Route> availableRoutes) {
         availableButtons.clear();
         for (Route route : availableRoutes) {
-            viewCallback.addClickListeners(ROUTE_GROUP_MAP.get(route.getGroupId()));
+            int id = route.getGroupId();
```

Comment:
I don't think you really need to create this variable. The route should be accessed in the loop so you can pass the groupId directly

Reply:
I created the variable to avoid calling getGroupId twice, but I don't mind if we change it